providers/dynamodb/shedlock-provider-dynamodb2/src/main/java/net/javacrumbs/shedlock/provider/dynamodb2/DynamoDBLockProvider.java

```diff
@@ -78,19 +78,33 @@ public class DynamoDBLockProvider implements LockProvider {
     private final String hostname;
     private final DynamoDbClient dynamoDbClient;
     private final String tableName;
+    private final String partitionKeyName;
 
     /**
      * Uses DynamoDB to coordinate locks
      *
-     * @param dynamoDbClient
-     *            v2 of DynamoDB client
-     * @param tableName
-     *            the lock table name
+     * @param dynamoDbClient v2 of DynamoDB client
+     * @param tableName      the lock table name
      */
     public DynamoDBLockProvider(@NonNull DynamoDbClient dynamoDbClient, @NonNull String tableName) {
         this.dynamoDbClient = requireNonNull(dynamoDbClient, "dynamoDbClient can not be null");
         this.tableName = requireNonNull(tableName, "tableName can not be null");
         this.hostname = Utils.getHostname();
+        this.partitionKeyName = ID;
```

Comment:
I think this should come from some config rather than constant if we have to keep it as keep it as custom PK

Reply:
I've added a new constructor to support the partitionKeyName argument. To maintain compatibility with the previous constructor, I’ve assigned partitionKeyName a default value of ID.
And this constant ID is only for dynamoDb provider. There is no any config file to keep this value.

Reply:
In that case, keeping ID variable makes no sense as it is not in use. We can directly refer the value as `_id`

Reply:
ID is being used in test case files.

Reply:
okay. We need to take up that in upcoming change if we will modify TC.